File: 2019/day1.py
#Part 1 (Separated because can't run through same file twice)
print (sum([(int(line)//3)-2 for line in open('day1.txt','r')]))
#Part 2
with open('day1.txt', 'r') as infile:
	fuel = []
	for line in infile:
		temp_fuel = int(line)
		fuel_sum = 0
		while fuel_num > 0:
			temp_fuel = max(0, int(fuel_num//3)-2)
			fuel_sum += fuel_num
		fuel.append(fuel_sum)
	print(sum(fuel))

File: 2019/day1.rb
lines = File.readlines("day1.txt").map(&:to_i)
operation = Proc.new {|n| n / 3 - 2}
puts lines.map(&operation).sum #part 1

total = []
lines.each do |n|
	while n > 0
		n = operation.call(n)
		n > 0 ? total << n : total << 0
	end
end
puts total.sum #part 2